benchmark/pkg/util/util.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"text/tabwriter"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+// MetricKey is used to identify a metric uniquely.
+type MetricKey struct {
+	TestName		string  // Name of the test ("Load Capacity", "Density", etc)
+	Verb			string  // "GET","LIST",etc for API calls and "POD STARTUP" for pod startup
+	Resource		string  // "nodes","pods", etc for API calls and empty value for pod startup
+	Percentile		string  // The percentile string ("Perc50", "Perc90", etc)
+}
+
+// MetricComparisonData holds all the values corresponding to a metric's comparison.
+type MetricComparisonData struct {
+	LeftJobSample	[]float64  // Sample values from the left job's runs
+	RightJobSample	[]float64  // Sample values from the right job's runs
+	Matched			bool       // Boolean indicating if the samples matched
+	Comments		string     // Any comments wrt the matching (for human interpretation)
+
+	// Below are some common statistical measures, that we would compute for the left
+	// and right job samples. They are used by some comparison schemes.
+	AvgL, AvgR		float64  // Average
+	StDevL, StDevR	float64  // Standard deviation
+	MaxL, MaxR		float64  // Max value
+}
+
+// JobComparisonData is a struct holding a map with keys as the metrics' keys and
+// values as their comparison data.
+type JobComparisonData struct {
+	Data			map[MetricKey]*MetricComparisonData
+}
+
+// Constructor for JobComparisonData struct.
+func NewJobComparisonData() *JobComparisonData {
+	return &JobComparisonData{
+		Data: make(map[MetricKey]*MetricComparisonData),
+	}
+}
+
+// Prints the job comparison data in a matrix form.
+func (j *JobComparisonData) PrettyPrint() {
+	const padding = 2
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, padding, ' ', 0)
+	fmt.Fprintf(w, "E2E TEST\tVERB\tRESOURCE\tPERCENTILE\tMATCHED?\tCOMMENTS\n")
```

Comment:
Yup. I don't care what's written and how, but mostly where. And we should be consistent about that.